### PR TITLE
Refactor AddressValidator

### PR DESCRIPTION
**Pull request by eminnett**
In an effort to remove duplication in the validators, we would like to move duplicated code into the ValidatesPresence concern and BaseValidator class. 

Future refactoring efforts will include the creation of the ValidatesLength concern.

This PR is a sample of the intended refactoring. We decided it would be easier to review the changes if they are applied to a single validator before applying the pattern to the rest of the validators.

### CODE REVIEW

File: app/validators/concerns/waste_exemptions_engine/validates_presence.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module WasteExemptionsEngine
+  module ValidatesPresence
```
Comment (Cruikshanks) [inline app/validators/concerns/waste_exemptions_engine/validates_presence.rb]: It's a name thing! If you look in `app/models/concerns/waste_exemptions_engine` you'll see we have a convention of naming our concerns `CanDoThing`. So in this case something along the lines of `CanValidatePresence` would bring it into line with that convention.

File: spec/support/shared_examples/validators/presence_validator.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples "a presence validator" do |validatable_class, property|
+  # it "includes ValidatesPresence" do
```
Comment (Cruikshanks) [inline spec/support/shared_examples/validators/presence_validator.rb]: Another convention we have is not to leave commented out code in the code base.

> I should really write these down somewhere 😳

Comment (eminnett) [reply spec/support/shared_examples/validators/presence_validator.rb]: This was an oversight.  It shouldn't have been commented out.

File: app/validators/concerns/waste_exemptions_engine/validates_presence.rb (hunk 1 above)
Comment (eminnett) [reply app/validators/concerns/waste_exemptions_engine/validates_presence.rb]: I'll fix this now.

File: spec/support/shared_examples/validators/presence_validator.rb (hunk 2 above)
Comment (eminnett) [reply spec/support/shared_examples/validators/presence_validator.rb]: Fixed in bcdf764.

File: app/validators/concerns/waste_exemptions_engine/validates_presence.rb (hunk 1 above)
Comment (eminnett) [reply app/validators/concerns/waste_exemptions_engine/validates_presence.rb]: Fixed in bcdf764

**Pull request merged**